client/src/components/ProductOverview/ImageGallery.jsx

```diff
@@ -1,19 +1,79 @@
 // import React, { useState } from 'react';
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
+import OverlayThumbnail from './OverlayThumbnail';
 
 function ImageGallery(props) {
-  const { productStyleSelected } = props;
+  const { productStyleSelected, expanded, handleExpand } = props;
+  const [mainPicUrl, setMainPicUrl] = useState(productStyleSelected.photos[0].url);
+  const [currIndex, setCurrIndex] = useState(0);
+
+  // On thumbnail click, set main image to thumbnail's image and current index to thumbnail's index
+  const selectMainPic = (overlayThumbnail) => {
+    setMainPicUrl(overlayThumbnail.url);
+    for (let j = 0; j < productStyleSelected.photos.length; j += 1) {
+      // console.log(productStyleSelected.photos[j]);
+      if (productStyleSelected.photos[j].url === overlayThumbnail.url) {
+        setCurrIndex(j);
+        break;
+      }
+    }
+  };
+
+  // To show next picture, up the current index and our hook will handle the render
+  const showNextPic = () => {
+    // console.log('index before showNextPic', currIndex, length);
+    if (currIndex !== productStyleSelected.photos.length - 1) {
+      setCurrIndex((prevState) => prevState + 1);
+      // console.log('index after showNextPic', currIndex);
+    }
+  };
+
+  // To show previous picture, lower the current index and our hook will handle the render
+  const showPrevPic = () => {
+    // console.log('index before showPrevPic', currIndex, length);
+    if (currIndex !== 0) {
+      setCurrIndex((prevState) => prevState - 1);
+      // console.log('index after showPrevPic', currIndex);
+    }
+  };
+
+  // Every time a new style is selected, reset main image and current index
```

Comment:
There must be a way to consolidate this into one UseEffect? Can't think of one now, but something to keep in mind for later later.